providers/google/src/airflow/providers/google/cloud/triggers/dataproc.py

```diff
@@ -150,23 +153,27 @@ def get_task_instance(self, session: Session) -> TaskInstance:
     async def get_task_state(self):
         from airflow.sdk.execution_time.task_runner import RuntimeTaskInstance
 
-        task_states_response = await sync_to_async(RuntimeTaskInstance.get_task_states)(
-            dag_id=self.task_instance.dag_id,
-            task_ids=[self.task_instance.task_id],
-            run_ids=[self.task_instance.run_id],
-            map_index=self.task_instance.map_index,
-        )
-        try:
-            task_state = task_states_response[self.task_instance.run_id][self.task_instance.task_id]
-        except Exception:
-            raise AirflowException(
-                "TaskInstance with dag_id: %s, task_id: %s, run_id: %s and map_index: %s is not found",
-                self.task_instance.dag_id,
-                self.task_instance.task_id,
-                self.task_instance.run_id,
-                self.task_instance.map_index,
+        if not self.task_instance:
+            raise AirflowException(f"TaskInstance not set on {self.__class__.__name__}!")
+
+        if not isinstance(self.task_instance, RuntimeTaskInstance):
+            task_states_response = await sync_to_async(RuntimeTaskInstance.get_task_states)(
+                dag_id=self.task_instance.dag_id,
+                task_ids=[self.task_instance.task_id],
+                run_ids=[self.task_instance.run_id],
+                map_index=self.task_instance.map_index,
             )
-        return task_state
+            try:
+                return task_states_response[self.task_instance.run_id][self.task_instance.task_id]
+            except Exception:
+                raise AirflowException(
```

Comment:
```suggestion
                raise RuntimeError(
```

not sure whehter we can duplicate these parts by moving them to some base class or mix-in. but that can be done in a following PR